src/menu/menu.service.ts

```diff
@@ -39,27 +39,23 @@ export class MenuService {
       }),
     ]);
 
-    if (!modifications || modifications?.length === 0) {
+    if (modificationGroups?.length === 0) {
       return menuItem;
     }
 
-    const modificationWithMenu =
-      await this.prisma.menuItemModification.createMany({
-        data: modifications.map((modification) => ({
-          menuItemId: menuItem.id,
-          modificationId: modification.modificationId,
-          price: modification.price,
-        })),
-      });
+    await this.prisma.menuItemModificationGroup.createMany({
+      data: modificationGroups.map((modification) => ({
+        menuItemId: menuItem.id,
+        modificationId: modification,
+      })),
+    });
 
-    return modificationWithMenu;
+    return 'created menu with modification';
   }
 
-  async createModification(dto: CreateModificationDto) {
-    const { modificationGroups } = dto;
-
+  async createModificationGroup(modificationGroups: ModificationGroupDto[]) {
     for (const modificationGroup of modificationGroups) {
-      const modification = await this.prisma.modificationGroup.create({
+      const modificationGroupItem = await this.prisma.modificationGroup.create({
```

Comment:
I see that you named it modificationGroupItem just because there was a naming collision. Do not do so. If you cannot come up with good naming yourself, ask GPT. because naming is what makes code manageable in future. bad naming ruins code little by little until it is impossible to manage. So, take your time to name variables. 

I understand that you want to focus on business logic and flow. In such instances, I usually do like you did, name variable as I want and focus on business logic. But before making commit, I refactor the code. I refactor naming, I refactor big functions and etc. Make it work -> make the code look good -> test that it still works.

Reply:
In your case, the function is not doing what it says it would do. 

createModificationGroup - creates modification group. But in your case, it creates many modification groups. 

Let's remove logic of creating many modification groups, because we create them one by one